pkg/volume/quobyte/quobyte.go

```diff
@@ -158,6 +158,12 @@ func (plugin *quobytePlugin) newMounterInternal(spec *volume.Spec, pod *api.Pod,
 		return nil, err
 	}
 
+	// We use the service account if it's not the default service account
+	if pod != nil && len(pod.Spec.ServiceAccountName) != 0 && pod.Spec.ServiceAccountName != "default" {
```

Comment:
the default service account shouldn't behave differently than other service accounts. I'm also not seeing how this adds protection in a shared env... the user has just as much control over the pod spec serviceAccountName as they do over the volume source user/group